### PR TITLE
[Tune][CI] Skip zoopt invalid values test

### PR DESCRIPTION
**Pull request by justinvyu**
Skips a zoopt searcher test that's causing the `test_searchers` suite in CI to be flaky. Skipping as this is not a Tune issue and needs to be fixed in the zoopt library.

## Why are these changes needed?

I re-enabled this test in https://github.com/ray-project/ray/pull/31147 since I thought that nan/inf error handling had been fixed in a recent zoopt release. However, nan/inf values will still cause an error if reported on trial complete.

This test should be enabled after https://github.com/polixir/ZOOpt/pull/7 is included in the next zoopt release.

## Related issue number

See https://github.com/ray-project/ray/issues/31439 and https://github.com/ray-project/ray/issues/31336
## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/tests/test_searchers.py [hunk 1]
```diff
@@ -327,9 +327,13 @@ def testSkopt(self):
...
 
     def testZOOpt(self):
+        self.skipTest(
+            "Recent ZOOpt versions fail handling invalid values gracefully. "
+            "Skipping until a fix is added in a future ZOOpt release."
+        )
         from ray.tune.search.zoopt import ZOOptSearch
 
-        np.random.seed(1000)  # At least one nan, inf, -inf and float
+        np.random.seed(1002)  # At least one nan, inf, -inf and float
```
Comment (bveeramani) [inline python/ray/tune/tests/test_searchers.py:336]: OOC why're we changing the seed?

File: python/ray/tune/tests/test_searchers.py [hunk 2]
```diff
@@ -327,9 +327,13 @@ def testSkopt(self):
         self.assertCorrectExperimentOutput(out)
 
     def testZOOpt(self):
+        self.skipTest(
+            "Recent ZOOpt versions fail handling invalid values gracefully. "
+            "Skipping until a fix is added in a future ZOOpt release."
+        )
```
Comment (bveeramani) [inline python/ray/tune/tests/test_searchers.py:333]: To ensure we eventually re-enable this test, should we create an issue to track this?

**Review by bveeramani — APPROVED** [review]
LGTM! Thanks!

File: python/ray/tune/tests/test_searchers.py (hunk 1 above)
Comment (justinvyu) [reply python/ray/tune/tests/test_searchers.py:336]: The test was passing for me locally with the old seed. I changed the seed and upped the number of samples to get the test failing on my local machine.

Comment (bveeramani) [reply python/ray/tune/tests/test_searchers.py:336]: Ah, okay. Makes sense

Comment (krfricke) [reply python/ray/tune/tests/test_searchers.py:336]: Does that seed produce one nan, inf, -inf and float? 
Which combination makes it fail?
Let's add this as a comment

**Review by krfricke — APPROVED** [review]
Thanks, let's document this and add an issue to track the revert

File: python/ray/tune/tests/test_searchers.py (hunk 1 above)
Comment (justinvyu) [reply python/ray/tune/tests/test_searchers.py:336]: Yes, this seed also produces at at least one of each. The failure happens when enough training samples have been collected and then a nan shows up and zoopt tries to replace one of its existing solutions with the nan solution. The previous seed only included a nan solution while it was still collecting training data. Will add as a comment!

**Pull request merged**